### DISCUSSION TITLE
Add IsInit in IDL for instruction accounts that are being initialised

### DISCUSSION THREAD

**Discussion by anoushk1234** [otter-sec/anchor#2356] [Ideas]
Consider the following as an example from the wordcel program for the ix createPost. Here profile is an account thats only being read and post is an account thats being initialised but we don't know this from looking at the IDL. Adding a field `isInit: true` would fix that. 

Would love to know the thoughts from the core contributors @armaniferrante @Henry-E  🙏

```
{
      "name": "createPost",
      "accounts": [
        {
          "name": "profile",
          "isMut": false,
          "isSigner": false
        },
        {
          "name": "post",
          "isMut": true,
          "isSigner": false
        },
        {
          "name": "authority",
          "isMut": true,
          "isSigner": true
        },
        {
          "name": "systemProgram",
          "isMut": false,
          "isSigner": false
        }
      ],
      "args": [
        {
          "name": "metadataUri",
          "type": "string"
        },
        {
          "name": "randomHash",
          "type": {
            "array": ["u8", 32]
          }
        }
      ]
    },
```

**Comment by Henry-E** [comment]
armani described possibly doing this in a github issue https://github.com/coral-xyz/anchor/issues/15 not sure if it ever really gained traction. I'm tempted to shutdown this discussion area and just stick with github issues tbh.

**Reply by anoushk1234** [reply]
@Henry-E i agree the discussions section is not very useful, can just use issues
